indexer/services/scripts/src/print-block.ts

```diff
@@ -0,0 +1,169 @@
+  });
+
+  logger.info({
+    at: 'consumer#seek',
+    message: 'Seeked.',
+    offset: offset.toString(),
+  });
+}
+
+export async function connect(height: number): Promise<void> {
+  await Promise.all([
+    consumer.connect(),
+    producer.connect(),
+  ]);
+
+  await consumer.subscribe({
+    topic: TO_ENDER_TOPIC,
+    fromBeginning: true,
+  });
+
+  addOnMessageFunction((_topic: string, message: KafkaMessage): Promise<void> => {
+    return printMessageAtHeight(message, height);
+  });
+
+  logger.info({
+    at: 'consumers#connect',
+    message: 'Connected to Kafka',
+  });
+}
+
+export async function printMessageAtHeight(
+  currentMessage: KafkaMessage,
+  targetHeight: number,
+): Promise<void> {
+  const indexerTendermintBlock: IndexerTendermintBlock | undefined = getIndexerTendermintBlock(
+    currentMessage,
+  );
+  if (indexerTendermintBlock === undefined) {
+    return;
+  }
+
+  const currentBlockHeight: number = parseInt(indexerTendermintBlock.height.toString(), 10);
+  if (currentBlockHeight < targetHeight) {
+    const offsetToSeek: number = targetHeight - currentBlockHeight + Number(currentMessage.offset);
+    await seek(BigInt(offsetToSeek));
+  } else if (currentBlockHeight === targetHeight) {
+    const annotatedEvents: AnnotatedIndexerTendermintEvent[] = [];
+    _.forEach(indexerTendermintBlock.events, (event: IndexerTendermintEvent) => {
+      const annotatedEvent:
+      AnnotatedIndexerTendermintEvent | undefined = annotateIndexerTendermintEvent(
+        event,
+      );
+      if (annotatedEvent === undefined) {
+        logger.error({
+          at: 'printMessageAtHeight',
+          message: 'Failed to parse event',
+          event,
+        });
+        throw Error('Failed to parse event');
+      }
+      annotatedEvents.push(annotatedEvent);
+    });
+    const annotatedBlock: AnnotatedIndexerTendermintBlock = {
+      ...indexerTendermintBlock,
+      events: [],
+      annotatedEvents,
+    };
+    logger.info({
+      at: 'printMessageAtHeight',
+      message: 'Printing block',
+      block: annotatedBlock,
+    });
+    await stopConsumer();
+  }
+}
+
+async function startKafka(height: number): Promise<void> {
+  logger.info({
+    at: 'index#start',
+    message: `Starting in env ${config.NODE_ENV}`,
+  });
+
+  await connect(height);
+  await startConsumer();
+
+  logger.info({
+    at: 'index#start',
+    message: 'Successfully started',
+  });
+}
+
+process.on('SIGTERM', async () => {
+  logger.info({
+    at: 'index#SIGTERM',
+    message: 'Received SIGTERM, shutting down',
+  });
+  await stopConsumer();
+});
+
+async function start(height: number): Promise<void> {
+  logger.info({
+    at: 'index#start',
+    message: `Connecting to kafka brokers: ${config.KAFKA_BROKER_URLS}`,
+  });
+  await startKafka(height);
+  logger.info({
+    at: 'index#start',
+    message: `Successfully connected to kafka brokers: ${config.KAFKA_BROKER_URLS}`,
+  });
+}
+
+const args = yargs.options({
```

Comment:
nit: only if it's easy to do, can we also add support to just parse a block at a particular offset

Reply:
I don't think this is needed. I believe all debug logs in Ender that print out offset should also print out block height, so we can just use the block height.